internal/notifier/alertmanager.go

```diff
@@ -134,8 +137,11 @@ func (s *Alertmanager) Post(ctx context.Context, event eventv1.Event) error {
 		},
 	}
 
-	err := postMessage(ctx, s.URL, s.ProxyURL, s.CertPool, payload)
-
+	err := postMessage(ctx, s.URL, s.ProxyURL, s.CertPool, payload, func(request *retryablehttp.Request) {
+		if s.Token != "" {
+			request.Header.Add("Authorization", "Bearer "+s.Token)
+		}
+	})
```

Comment:
```suggestion
	var opts []requestOptFunc
	if s.Token != "" {
		opts = append(opts, func(request *retryablehttp.Request) {
			request.Header.Add("Authorization", "Bearer "+s.Token)
		})
	}
	err := postMessage(ctx, s.URL, s.ProxyURL, s.CertPool, payload, opts...)
```

Your original code would work correctly, but I believe putting the if outside the option makes more sense. In this suggestion, an option would only be passed if the token is not empty.